src/os/bluestore/BlueStore.cc

```diff
@@ -1750,7 +1758,9 @@ void BlueStore::BufferSpace::read(
     for (auto i = _data_lower_bound(offset);
          i != buffer_map.end() && offset < end && i->first < end; ++i) {
       Buffer *b = &i->second;
-      ceph_assert(b->end() > offset);
```

Comment:
The assert should remain here.

We assume no buffers overlap, and buffers are processed in order.
On first iteration, with i from `_data_lower_bound()` condition holds:
`offset < b->end()` but we cannot know whether `offset <= b->offset` or `offset > b->offset`
On each subsequent iteration conditions holds:
`offset <= b->offset` and `offset < b->end()`

So the assert checks our assumptions on buffers.